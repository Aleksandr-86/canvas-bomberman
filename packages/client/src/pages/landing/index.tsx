import { Link } from 'react-router-dom'

import styles from './landing.module.css'

export const Landing = () => {
  const isAuth = false
  const displayName = 'Алекс'

  let content: JSX.Element

  if (isAuth) {
    content = (
      <>
        <div className={styles.landing__description_top}>
          Привет {displayName}! <br />
          Спешим напомнить, что ты в любой момент можешь освежить свои знания по
          игре, перечитав{' '}
          <Link className={styles.landing__link} to="/rules">
            правила
          </Link>{' '}
          и посмотреть{' '}
          <Link className={styles.landing__link} to="/leaderboard">
            таблицу лидеров!
          </Link>
        </div>

        <Link to="/game">
          <button className={styles.landing__button}>Погнали</button>
        </Link>
      </>
    )
  } else {
    content = (
      <>
        <div className={styles.landing__description_top}>
          Привет, мы рады приветствовать тебя! <br />
          Прежде чем ты начнёшь взрывать всё вокруг, загляни в{' '}
          <Link className={styles.landing__link} to="/rules">
            правила
          </Link>
          . Уверены, это поможет тебе возглавить нашу{' '}
          <Link className={styles.landing__link} to="/leaderboard">
            таблицу лидеров!
          </Link>
        </div>

        <div className={styles['landing__button-wrapper']}>
          <Link to="/sign-in">
            <button className={styles.landing__button}>Войти</button>
          </Link>

          <Link to="/sign-up">
            <button className={styles.landing__button}>
              Зарегистрироваться
            </button>
          </Link>
        </div>
      </>
    )
  }

  return (
    <div className={styles.landing}>
      <div className={styles.landing__wrapper}>
        <img
          className={styles.landing__logo}
          src="src/assets/images/bomberman-logo.png"
          alt="bomberman-log"
        />
        {content}
        <div className={styles.landing__description_bottom}>
          Есть вопросы? Скорее переходи к нам на{' '}
          <Link className={styles.landing__link} to="/forum">
            форум
          </Link>
        </div>
      </div>
      <div className={styles['landing__img-wrapper']}>
        <img className="landing__hero-img" src="src/assets/images/hero.png" />
      </div>
    </div>
  )
}
