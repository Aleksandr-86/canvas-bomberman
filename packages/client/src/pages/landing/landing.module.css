.landing {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 100vh;
  user-select: none;
}

.landing__wrapper {
  display: flex;
  flex-direction: column;
  gap: 50px;
  align-items: center;
}

.landing__logo {
  bottom: 0;
  width: 1046px;
  height: 267px;
}

.landing__description_top,
.landing__description_bottom {
  width: 1260px;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
}

.landing__description_top {
  font-size: 40px;
  line-height: 47.4px;
}

.landing__description_bottom {
  font-size: 20px;
}

.landing__link {
  text-decoration: none;
}

.landing__button-wrapper {
  display: flex;
  gap: 50px;
  justify-content: center;
  align-items: center;
}

.landing__button {
  font-size: 40px;
  text-transform: uppercase;
}

.landing__img-wrapper {
  display: flex;
  justify-content: flex-end;
  width: 1446px;
}

.landing__hero-img {
  width: 253px;
  height: 191px;
}
