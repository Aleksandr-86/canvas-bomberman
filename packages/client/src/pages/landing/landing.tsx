import { Link } from 'react-router-dom'

import { useAppSelector } from '../../store/hooks'
import { getUser } from '../../store/selectors'
import { NavigationBar } from '../../components/navigation-bar/navigation-bar'

import baseStyles from '../../app/app.module.css'
import styles from './landing.module.css'

// Содержание для авторизованных пользователей
const ContentLogged = () => {
  const { displayName } = useAppSelector(getUser)

  return (
    <>
      <div className={styles.landing__description_top}>
        Привет {displayName}! <br />
        Спешим напомнить, что ты в любой момент можешь освежить свои знания по
        игре, перечитав{' '}
        <Link className={styles.landing__link} to="/rules">
          правила
        </Link>{' '}
        и посмотреть{' '}
        <Link className={styles.landing__link} to="/leaderboard">
          таблицу лидеров!
        </Link>
      </div>
      <Link className={baseStyles.link_button} to="/game">
        Погнали
      </Link>
    </>
  )
}

// Содержание для неавторизованных пользователей
const ContentNotLogged = () => (
  <>
    <div className={styles.landing__description_top}>
      Привет, мы рады приветствовать тебя! <br />
      Прежде чем ты начнёшь взрывать всё вокруг, загляни в{' '}
      <Link className={styles.landing__link} to="/rules">
        правила
      </Link>
      . Уверены, это поможет тебе возглавить нашу{' '}
      <Link className={styles.landing__link} to="/leaderboard">
        таблицу лидеров!
      </Link>
    </div>
    <div className={styles['landing__button-wrapper']}>
      <Link className={baseStyles.link_button} to="/sign-in">
        Войти
      </Link>
      <Link className={baseStyles.link_button} to="/sign-up">
        Зарегистрироваться
      </Link>
    </div>
  </>
)

export const Landing = () => {
  const { isAuth } = useAppSelector(state => state.user)

  const navBar = isAuth ? <NavigationBar /> : null
  const content = isAuth ? <ContentLogged /> : <ContentNotLogged />

  return (
    <div className={styles.landing}>
      {navBar}
      <div className={styles.landing__wrapper}>
        <img
          className={styles.landing__logo}
          src="src/assets/images/bomberman-logo.png"
          alt="bomberman-logo"
          draggable="false"
        />
        {content}
        <div className={styles.landing__description_bottom}>
          Есть вопросы? Скорее переходи к нам на{' '}
          <Link className={styles.landing__link} to="/forum">
            форум
          </Link>
        </div>
      </div>
      <img
        className={styles['landing__hero-img']}
        src="src/assets/images/hero.png"
        alt="hero-img"
        draggable="false"
      />
    </div>
  )
}
