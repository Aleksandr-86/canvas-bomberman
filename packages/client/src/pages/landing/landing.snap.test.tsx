import * as renderer from 'react-test-renderer'
import { Landing } from './landing'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'

const mockStore = configureStore([])
const store = mockStore({
  user: {
    isAuth: true,
    user: {
      displayName: `mockDisplayName`,
      login: `mockLogin`,
      firstName: `mockFirstName`,
      secondName: `mockSecondName`,
      email: `mockEmail`,
      phone: 8900000000,
    },
  },
  theme: {
    current: `mockLight`,
  },
})

describe(`<Landing/> snapshot test cases`, () => {
  test(`<Landing/> should render <div className=landing> parent with child elements (AUTH)`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter initialEntries={[`/`]}>
            <Landing />
          </MemoryRouter>
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`<Landing/> should render <div className=landing> parent with child elements (NO_AUTH)`, () => {
    const store = mockStore({
      user: { isAuth: false, user: { id: 1 } },
      theme: { current: `light` },
    })

    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter initialEntries={[`/`]}>
            <Landing />
          </MemoryRouter>
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
