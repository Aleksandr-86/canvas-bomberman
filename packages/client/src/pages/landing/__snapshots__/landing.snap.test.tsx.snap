// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing/> snapshot test cases <Landing/> should render <div className=landing> parent with child elements (AUTH) 1`] = `
<div
  className="landing"
>
  <nav
    className="navigationBar"
  >
    <img
      alt="user-avatar"
      draggable="false"
    />
    <a
      className="link"
      href="/profile"
      onClick={[Function]}
    >
      mockDisplayName
    </a>
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      Выйти
    </a>
  </nav>
  <div
    className="wrapper"
  >
    <img
      alt="bomberman-logo"
      className="logo"
      draggable="false"
    />
    <div
      className="descriptionTop"
    >
      Привет 
      mockDisplayName
      ! 
      <br />
      Спешим напомнить, что ты в любой момент можешь освежить свои знания по игре, перечитав
       
      <a
        className="link"
        href="/rules"
        onClick={[Function]}
      >
        правила
      </a>
       
      и посмотреть
       
      <a
        className="link"
        href="/leaderboard"
        onClick={[Function]}
      >
        таблицу лидеров!
      </a>
    </div>
    <a
      className="linkButton"
      href="/game"
      onClick={[Function]}
    >
      Погнали
    </a>
    <div
      className="descriptionBottom"
    >
      Есть вопросы? Скорее переходи к нам на
       
      <a
        className="link"
        href="/forum"
        onClick={[Function]}
      >
        форум
      </a>
    </div>
  </div>
  <img
    alt="hero-img"
    className="heroImg"
    draggable="false"
  />
</div>
`;

exports[`<Landing/> snapshot test cases <Landing/> should render <div className=landing> parent with child elements (NO_AUTH) 1`] = `
<div
  className="landing"
>
  <div
    className="wrapper"
  >
    <img
      alt="bomberman-logo"
      className="logo"
      draggable="false"
    />
    <div
      className="descriptionTop"
    >
      Привет, мы рады приветствовать тебя! 
      <br />
      Прежде чем ты начнёшь взрывать всё вокруг, загляни в
       
      <a
        className="link"
        href="/rules"
        onClick={[Function]}
      >
        правила
      </a>
      . Уверены, это поможет тебе возглавить нашу
       
      <a
        className="link"
        href="/leaderboard"
        onClick={[Function]}
      >
        таблицу лидеров!
      </a>
    </div>
    <div
      className="buttonWrapper"
    >
      <a
        className="linkButton"
        href="/sign-in"
        onClick={[Function]}
      >
        Войти
      </a>
      <button
        className="button buttonYandex"
        onClick={[Function]}
      >
        <img
          alt="Логотип Яндекса"
          width={40}
        />
        ID
      </button>
      <a
        className="linkButton"
        href="/sign-up"
        onClick={[Function]}
      >
        Зарегистрироваться
      </a>
    </div>
    <div
      className="descriptionBottom"
    >
      Есть вопросы? Скорее переходи к нам на
       
      <a
        className="link"
        href="/forum"
        onClick={[Function]}
      >
        форум
      </a>
    </div>
  </div>
  <img
    alt="hero-img"
    className="heroImg"
    draggable="false"
  />
</div>
`;
